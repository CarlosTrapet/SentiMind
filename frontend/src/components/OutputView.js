import React from 'react';
import SubHeader from './SubHeader';
import Sentiment from './Sentiment';
import Button from './Button';

const OutputView = props => {
  return (
    <div>
      <SubHeader subHeaderText="So, the world thinks..." />
      <Sentiment />
      <Button buttonText="Choose another topic" handleSubmit={props.handleSubmit} />
    </div>
  );
};

export default OutputView;
