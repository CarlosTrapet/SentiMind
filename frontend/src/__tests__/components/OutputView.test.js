import React from 'react';
import { shallow } from 'enzyme';
import OutputView from '../../components/OutputView';

describe('OutputView', () => {
  const mockHandleSubmit = jest.fn();
  const props = {
    handleSubmit: mockHandleSubmit,
    average_sentiment: 'negative',
    query: 'Donald Trump',
    sentiments: { positive: 0, neutral: 0, negative: 0 },
    top_tweets: { positive: '102389292', negative: '02392039420' },
    top_words: { hello: 4, goodbye: 3, again: 2 }
  };
  const outputView = shallow(<OutputView {...props} />);

  it('renders correctly', () => {
    expect(outputView).toMatchSnapshot();
  });

  it('renders subheader component', () => {
    expect(outputView.find('SubHeader').exists()).toBe(true);
  });

  it('renders sentiment component', () => {
    expect(outputView.find('Sentiment').exists()).toBe(true);
  });

  it('renders Button component', () => {
    expect(outputView.find('Button').exists()).toBe(true);
  });

  it('renders a BarChart component', () => {
    expect(outputView.find('VictoryChart').exists()).toBe(true);
  });

  describe('tweets', () => {
    it('renders a TwitterTweetEmbed component for positive tweet', () => {
      expect(outputView.find('#positiveTweet').exists()).toBe(true);
    });

    it('renders a TwitterTweetEmbed component for negative tweet', () => {
      expect(outputView.find('#negativeTweet').exists()).toBe(true);
    });
  });

  describe('passing props', () => {
    describe('SubHeader props', () => {
      it('passes flexible SubHeaderText down to main-subheader', () => {
        expect(outputView.find('#main-subheader').prop('subHeaderText')).toBe(
          'The weighted sentiment on "Donald Trump" is:'
        );
      });

      it('passes flexible SubHeaderText down to positiveTweet subheader', () => {
        expect(
          outputView.find('#positiveTweet-subheader').prop('subHeaderText')
        ).toBe('Most Positive Tweet');
      });

      it('passes flexible SubHeaderText down to negativeTweet subheader', () => {
        expect(
          outputView.find('#negativeTweet-subheader').prop('subHeaderText')
        ).toBe('Most Negative Tweet');
      });
    });

    describe('Sentiment props', () => {
      it('passes sentiment down to Sentiment', () => {
        expect(outputView.find('Sentiment').prop('average_sentiment')).toBe(
          'negative'
        );
      });
    });

    describe('Button props', () => {
      it('passes flexible buttonText down to Button', () => {
        expect(outputView.find('Button').prop('buttonText')).toBe(
          'Choose another topic'
        );
      });
      it('handle submit function down to Button', () => {
        expect(outputView.find('Button').prop('handleSubmit')).toBe(
          mockHandleSubmit
        );
      });
    });

    describe('formatTopWords()', () => {
      it('correctly formats json data', () => {
        const json_data = { hello: 4, goodbye: 3, again: 2 };
        const formatted_data = [
          { text: 'hello', value: 400 },
          { text: 'goodbye', value: 300 },
          { text: 'again', value: 200 }
        ];
        expect(outputView.instance().formatTopWords(json_data)).toEqual(
          formatted_data
        );
      });
    });
  });
});
