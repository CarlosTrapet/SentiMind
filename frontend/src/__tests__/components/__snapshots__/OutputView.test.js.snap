// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutputView renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OutputView
    handleSubmit={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SubHeader
          subHeaderText="So, the world thinks..."
/>,
        <Sentiment />,
        <Button
          buttonText="Choose another topic"
          handleSubmit={[Function]}
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "subHeaderText": "So, the world thinks...",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "buttonText": "Choose another topic",
          "handleSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SubHeader
            subHeaderText="So, the world thinks..."
/>,
          <Sentiment />,
          <Button
            buttonText="Choose another topic"
            handleSubmit={[Function]}
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "subHeaderText": "So, the world thinks...",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "buttonText": "Choose another topic",
            "handleSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
